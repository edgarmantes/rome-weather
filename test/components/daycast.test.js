import React from 'react'
import renderer from 'react-test-renderer';
import DayCast from '../../src/components/daycast';

describe('<DayCast />' , () => {


	let wrapper

	beforeEach( ()=>{
		wrapper = shallow(<DayCast day={"Friday"} hi={80} lo={60} icon={"something"}  />)
	})

	it('Renders DayCast component properly', () => {
		expect(wrapper.length).toEqual(1)
	});

	it('Adds Day passed in value to this.props.day', () => {
		expect(wrapper.find(".five-day").text()).toEqual("Friday")
	})

	it('Adds Date passed in value to this.props.hi', () => {
		let temp = "hi: 80°"
		expect(wrapper.find(".five-hi").text()).toEqual(temp)
	})

	it('Adds Date passed in value to this.props.lo', () => {
		let temp = "lo: 60°"
		expect(wrapper.find(".five-lo").text()).toEqual(temp)
	})

	it('Snapshot of DayCast', () => {
        const renderedValue =  renderer.create(<DayCast />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
