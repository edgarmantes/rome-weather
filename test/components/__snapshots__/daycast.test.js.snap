// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayCast /> Snapshot of DayCast 1`] = `
<li
  className={undefined}
>
  <span
    className="five-day"
  />
  <div
    className="five-temps"
  >
    <p
      className="five-hi"
    >
      hi: 
      °
    </p>
    <p
      className="five-lo"
    >
      lo: 
      °
    </p>
  </div>
  <div
    className="five-img-cont"
  >
    <img
      className="five-img"
      src="http://openweathermap.org/img/w/undefined.png"
    />
  </div>
</li>
`;
